File: VITRAL/vitral/src/vitral/toolkits/environment/SimpleBackground.java
//===========================================================================
//=-------------------------------------------------------------------------=
//= Module history:                                                         =
//= - November 27 2005 - Oscar Chavarro: Original base version              =
//===========================================================================

package vitral.toolkits.environment;

import vitral.toolkits.common.ColorRgb;
import vitral.toolkits.common.Vector3D;

public class SimpleBackground extends Background {
    ColorRgb _color;

    public SimpleBackground() {
        super();

        _color = new ColorRgb();
        _color.r = 0;
        _color.g = 0;
        _color.b = 0;
    }

    public ColorRgb colorInDireccion(Vector3D d)
    {
        return _color;
    }

    public void setColor(double r, double g, double b)
    {
        _color.r = r;
        _color.g = g;
        _color.b = b;
    }
}

//===========================================================================
//= EOF                                                                     =
//===========================================================================

File: typescript/base/src/main/vsdk/toolkit/environment/background/SimpleBackground.ts
import { ColorRgb } from "../../common/color/ColorRgb.js";
import { Vector3Dd } from "../../common/linealAlgebra/Vector3Dd.js";
import { Background } from "./Background.js";

/** A uniform, direction-independent background colour. */
export class SimpleBackground extends Background {
    private color = new ColorRgb(0, 0, 0);

    public override colorInDireccion(_d: Vector3Dd): ColorRgb {
        return new ColorRgb(this.color);
    }

    public setColor(r: number, g: number, b: number): void {
        this.color = new ColorRgb(r, g, b);
    }
}
